### PR TITLE
Refs #22620 - provide repo types on rootScope

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://github.com/Katello/katello/pull/7204 makes use of this

File: app/views/bastion/layouts/assets.html.erb [hunk 1]
```diff
@@ -20,6 +20,7 @@
     angular.module('Bastion').value('CurrentOrganization', "<%= Organization.current.id if Organization.current %>");
     angular.module('Bastion').value('contentAccessMode', "<%= Organization.current.try(:content_access_mode) if Organization.current %>");
     angular.module('Bastion').value('foreman', tfm);
+    angular.module('Bastion').value('repositoryTypes', angular.fromJson('<%= Katello::RepositoryTypeManager.repository_types.values.to_json.html_safe %>'));
```
Comment (waldenraines) [inline app/views/bastion/layouts/assets.html.erb]: This change makes it seem even more odd that bastion and katello are separate as it sets a new precedent by introducing katello concepts into bastion.  Really make me want to combine bastion and katello some day.

File: app/assets/javascripts/bastion/bastion.module.js [hunk 2]
```diff
@@ -76,18 +76,27 @@ angular.module('Bastion').config(
...
-    function ($rootScope, $state, $stateParams, gettextCatalog, currentLocale, $location, $window, $breadcrumb, PageTitle, BastionConfig) {
+angular.module('Bastion').run(['$rootScope', '$state', '$stateParams', 'gettextCatalog', 'currentLocale', '$location', '$window', '$breadcrumb', 'PageTitle', 'BastionConfig', 'repositoryTypes',
+    function ($rootScope, $state, $stateParams, gettextCatalog, currentLocale, $location, $window, $breadcrumb, PageTitle, BastionConfig, repositoryTypes) {
         var fromState, fromParams, orgSwitcherRegex;
 
         $rootScope.$state = $state;
         $rootScope.$stateParams = $stateParams;
         $rootScope.transitionTo = $state.transitionTo;
         $rootScope.$location = $location;
+        $rootScope.$repositoryTypes = repositoryTypes;
```
Comment (waldenraines) [inline app/assets/javascripts/bastion/bastion.module.js]: I know I told you on IRC that I was fine with this but now that I see it I'm having a bad reaction.  I know that all of the katello angular stuff is deprecated but I wonder if we should set this precedent. 

Using a service and then injecting that service into each controller is more work but looking at https://github.com/Katello/katello/pull/7204 there are only a handful of places where this is necessary.  When this change was described to me I thought there were going to be many more places where the DI would be necessary.

Comment (jlsherrill) [reply app/assets/javascripts/bastion/bastion.module.js]: yeah, it ended up being fewer places than i intiailly thought.  I'm okay moving it to a service.

Comment (ehelms) [reply app/assets/javascripts/bastion/bastion.module.js]: Given the above, I'd vote service too. I agree if this is deprecated code we don't need to care as much but given we still have to maintain it we don't want it to fall into complete disarray given bugs can pop up from anywhere.

File: app/views/bastion/layouts/assets.html.erb (hunk 1 above)
Comment (jlsherrill) [reply app/views/bastion/layouts/assets.html.erb]: we've already done this with katello specific settings.

Comment (jlsherrill) [reply app/views/bastion/layouts/assets.html.erb]: such as unregister_delete_host

Comment (waldenraines) [reply app/views/bastion/layouts/assets.html.erb]: True, I'm not saying I'm against the change but that I'm in favor of merging bastion and katello.

**Comment by jlsherrill** [comment]
@waldenraines okay with this?

**Comment by jlsherrill** [comment]
@waldenraines do you mind doing a release for me?

**Comment by waldenraines** [comment]
> @waldenraines do you mind doing a release for me?

Just released 6.1.9

**Pull request merged**